src/main/java/org/opentripplanner/updater/bike_rental/GenericJsonBikeRentalDataSource.java

```diff
@@ -84,20 +114,27 @@ public boolean update() {
             }
             // TODO handle optional GBFS files, where it's not warning-worthy that they don't exist.
             if (data == null) {
-                log.warn("Failed to get data from url " + url);
+                addError("Failed to get data from url " + url);
                 return false;
             }
             parseJSON(data);
-            data.close();
         } catch (IllegalArgumentException e) {
-            log.warn("Error parsing bike rental feed from " + url, e);
+            addError("Error parsing bike rental feed from " + url, e);
             return false;
         } catch (JsonProcessingException e) {
-            log.warn("Error parsing bike rental feed from " + url + "(bad JSON of some sort)", e);
+            addError("Error parsing bike rental feed from " + url + "(bad JSON of some sort)", e);
```

Comment:
Should this use String.format instead?

Reply:
It didn't before and I hope this is refactored away anyways in the GBFS 2.2 update, so it doesn't seem worth the effort.